collect_app/src/test/java/org/odk/collect/android/widgets/PrinterWidgetTest.kt

```diff
@@ -0,0 +1,76 @@
+package org.odk.collect.android.widgets
+
+import com.google.android.material.button.MaterialButton
+import org.hamcrest.CoreMatchers.equalTo
+import org.hamcrest.MatcherAssert.assertThat
+import org.javarosa.core.model.data.IAnswerData
+import org.junit.Test
+import org.mockito.kotlin.any
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.verifyNoInteractions
+import org.mockito.kotlin.whenever
+import org.odk.collect.android.R
+import org.odk.collect.android.formentry.PrinterWidgetViewModel
+import org.odk.collect.android.formentry.questions.QuestionDetails
+import org.odk.collect.android.support.WidgetTestActivity
+import org.odk.collect.android.utilities.QuestionMediaManager
+import org.odk.collect.android.widgets.base.QuestionWidgetTest
+import org.odk.collect.printer.HtmlPrinter
+import org.odk.collect.qrcode.QRCodeCreator
+import org.odk.collect.testshared.FakeScheduler
+
+class PrinterWidgetTest : QuestionWidgetTest<PrinterWidget, IAnswerData>() {
```

Comment:
I'd really like to move away from `QuestionWidgetTest` and use shared helpers rather than extension. One example of why would be the no-op tests we end up with here. Do you think it'd be ok to write a version of `PrinterWidgetTest` that doesn't extend `QuestionWidgetTest` or would you rather we just deprecate `QuestionWidgetTest` in follow-up PR?

Reply:
Hmm I'm not sure if we introduced a concept of non-question form elements like this new widget or the URL widget etc then it would make some sense to rework it but now ok in this case it doesn't look good maybe but forcing writing some tests when we add anew widget generally is a useful thing. Please file a separate issue for that.